src/identity/util/FetchUtil.ts

```diff
@@ -14,15 +14,15 @@ export async function fetchDataset(url: string): Promise<Dataset> {
   try {
     rawResponse = (await fetch(url)) as DatasetResponse<Dataset>;
   } catch (err: unknown) {
-    const errorMessage = `Cannot fetch ${url}: ${isNativeError(err) ? err.message : 'Unknown error'}`;
+    const errorMessage = `Cannot fetch ${url}: ${createErrorMessage(err)}`;
     logger.error(errorMessage);
     throw new Error(errorMessage);
   }
   let dataset: Dataset;
   try {
     dataset = await rawResponse.dataset();
   } catch (err: unknown) {
-    const errorMessage = `Could not parse RDF in ${url}: ${isNativeError(err) ? err.message : 'Unknown error'}`;
+    const errorMessage = `Could not parse RDF in ${url}: ${createErrorMessage(err)}`;
```

Comment:
Possible security vulnerability here, when the server would be forced to read arbitrary files from its on intranet. Let's log the error message but not echo it.